### PR TITLE
Add unit test report

### PR DESCRIPTION
**Pull request by Herman0802**

### CODE REVIEW

File: rookpoly/board.py [hunk 1]
```diff
@@ -7,90 +7,108 @@
...
 
     def get_empty_squares(self):
-        for (row, i) in enumerate(self.board):
-            for (column, j) in enumerate(i):
-                if j == '0':
-                    yield row, column
-
-    def get(self, i, j):
-        return self.board[i][j]
+        """Yield the positions of empty squares on the board."""
```
Comment (DelevoXDG) [inline rookpoly/board.py:71]: Nit: I'm not sure "yield" is a best way to describe this

File: tests/.DS_Store
Comment (DelevoXDG) [inline tests/.DS_Store:1]: Please remove this. You may also add it to .gitignore

File: tests/test_polynomial.py [hunk 2]
```diff
@@ -5,141 +5,90 @@
 
 class TestPolynomial(unittest.TestCase):
 
+    def setUp(self):
+        self.p1 = Polynomial(1, 2, 3)
+        self.p2 = Polynomial(3, 2, 1)
+        self.scalar = 5
+        self.zero_poly = Polynomial()
```
Comment (DelevoXDG) [inline tests/test_polynomial.py:12]: Why this change? The premise of unit tests is to be isolated, I don't really see the need for setUp there

File: tests/test_polynomial.py [hunk 3]
```diff
@@ -5,141 +5,90 @@
...
-        self.assertEqual(p.coeffs, [1, 2, 3])
+        # Test various ways of initializing a Polynomial
+        self.assertEqual(Polynomial(1, 2, 3).coeffs, [1, 2, 3])
+        self.assertEqual(Polynomial([1, 2, 3]).coeffs, [1, 2, 3])
+        self.assertEqual(Polynomial(self.p1).coeffs, self.p1.coeffs)
+        self.assertEqual(Polynomial(5).coeffs, [5])
+        self.assertEqual(Polynomial().coeffs, [])
+        self.assertEqual(Polynomial([1, 2, 3, 0, 0, 0]).coeffs, [1, 2, 3])
 
-        # Test initialization from scalars
```
Comment (DelevoXDG) [inline tests/test_polynomial.py:13]: Why remove the comments related to specific way to initialize? I think that kind of reduces the value of comments, no?

File: tests/test_polynomial.py [hunk 4]
```diff
@@ -5,141 +5,90 @@
...
+    def test_call(self):
+        # Test polynomial evaluation at given values
+        self.assertEqual(self.p1(0), 1)
+        self.assertEqual(self.p1(1), 6)
+        self.assertEqual(self.p1(2), 17)
 
-        # Test initialization with single scalar
-        p = Polynomial(5)
-        self.assertEqual(p.coeffs, [5])
+    def test_eq(self):
```
Comment (DelevoXDG) [inline tests/test_polynomial.py:35]: Please don't unify the test cases. The unit tests should be as atomic as possible

File: tests/test_polynomial.py [hunk 5]
```diff
@@ -5,141 +5,90 @@
...
+        # Test right-hand addition (scalar + polynomial)
+        self.assertEqual((self.scalar + self.p1).coeffs, [6, 2, 3])
 
     def test_rmul(self):
-        # Test right-hand multiplication
-        p1 = Polynomial(1, 2, 3)
-        p2 = 5 * p1
-        self.assertEqual(p2.coeffs, [5, 10, 15])
+        # Test right-hand multiplication (scalar * polynomial)
+        self.assertEqual((self.scalar * self.p1).coeffs, [5, 10, 15])
```
Comment (DelevoXDG) [inline tests/test_polynomial.py:73]: This became significantly less readable. You don't need a whole set up do define an int variable. They are usually done for larger objects. I don't think there's any reason to reuse an int variable.

File: tests/test_report.md
Comment (DelevoXDG) [inline tests/test_report.md:1]: Very good text overall. Are you planning to add a piece of info about blackbox tests, or should I do that?

File: tests/test_report.md [hunk 6]
```diff
@@ -0,0 +1,82 @@
...
+| test_pow                     | Raise polynomial to a power (not implemented)         | PASS ✅   |
+
+## Analysis and Future Directions
+- **Consistency with Mathematical Theory**: The tests of both classes confirm their consistency with the principles of the chess polynomial, which is crucial for the credibility of the project.
+- **Development Opportunities**: Considering the implementation of additional functions and test scenarios, especially for the `Polynomial` class, may enrich the project's functionality.
+- **Recommendations**: It is recommended to conduct additional tests in more complex scenarios to fully utilize the potential of both classes in a mathematical context.
+
+## Summary
+The unit tests of the `Board` and `Polynomial` classes have successfully confirmed their correctness and effectiveness in the context of the RookPoly project. Both classes meet the mathematical requirements associated with the chess polynomial and provide a solid foundation for further research and development in the field of discrete mathematics. The test results provide strong foundations for further work on the project.
```
Comment (DelevoXDG) [inline tests/test_report.md:82]: 🤣 I think that's a bit of a stretch
```suggestion
The unit tests of the `Board` and `Polynomial` classes have successfully confirmed their correctness and effectiveness in the context of the RookPoly project. Both classes meet the mathematical requirements associated with the chess polynomial. The test results provide strong foundations for further work on the project.
```

File: tests/test_report.md [hunk 7]
```diff
@@ -0,0 +1,82 @@
+Herman Lazarchyk, Maksim Zdobnikau
+
+# Comprehensive Unit Test Report for the RookPoly Project
```
Comment (DelevoXDG) [inline tests/test_report.md]: It would then make sense to make rename this
```suggestion
# Comprehensive Test Report for the RookPoly Project
```

File: tests/test_polynomial.py (hunk 2 above)
Comment (Herman0802) [reply tests/test_polynomial.py:12]: The thing is that the same values were used in many places, so I decided to move these values to the `setUp` section

File: rookpoly/board.py (hunk 1 above)
Comment (Herman0802) [reply rookpoly/board.py:71]: At the end of the function, `yield` is used, so I decided to indicate this in its description

File: tests/test_polynomial.py (hunk 3 above)
Comment (Herman0802) [reply tests/test_polynomial.py:13]: All initialization methods are well described in the definition of the `Polynomial` class constructor, so listing them again, in my opinion, does not make sense

File: tests/test_report.md (hunk 6 above)
Comment (Herman0802) [reply tests/test_report.md:82]: I think it's okay to leave a pinch of pathos at the end :)

File: tests/test_report.md
Comment (Herman0802) [reply tests/test_report.md:1]: I think you are more suitable for this task, since you were directly involved in their development and are more immersed in the topic

File: tests/test_polynomial.py (hunk 4 above)
Comment (Herman0802) [reply tests/test_polynomial.py:35]: I think in the case of not very large tests, the chosen approach can be maintained
